src/components/widgets-settings/params-picker/params-picker.directive.coffee

```diff
@@ -14,16 +14,26 @@ module.controller('SettingParamsPickerCtrl', ($scope) ->
       'ui-floating': true,
       tolerance: 'pointer'
     }
-    setting.isInitialized = true if _.isEmpty($scope.options)
+    $scope.applyForDashboard = false;
+    $scope.applyForAll = ->
+      setting.reach = $scope.applyForDashboard;
 
+    setting.isInitialized = true if _.isEmpty($scope.options)
 
-  setting.toMetadata = ->
+  setting.toMetadata = ()->
     param = {}
     param[$scope.param] = _.compact(_.map $scope.options, (statusOption) ->
       statusOption.label if statusOption.selected
     )
+    param.reach = $scope.applyForDashboard = false;
     return param
 
+  setting.getOptions = ->
+    return $scope.options
+
+  setting.setOptions = (options)->
+    _.assign $scope.options, options
+
```

Comment:
"options" attribute in the setting should not be accessed from the outside of the directive as a setting is a stateless element that is initialized based on the widgets' metadata. Instead, the standard cycle should be followed:
  1. widgets' metadata are retrieved or changed
  2. WidgetsSvc.show is retrieving the widgets' data based on these metadata
  3. widget.initContext() is called to initialize the widget
  4. setting.initialize() is called for each setting of the widget